src/main/java/net/minestom/server/entity/Entity.java

```diff
@@ -483,7 +484,11 @@ public void tick(long time) {
                         getVelocity().getZ() / tps
                 );
 
-                this.onGround = CollisionUtils.handlePhysics(this, deltaPos, newPosition, newVelocityOut);
+                if (this.hasPhysics) {
+                    this.onGround = CollisionUtils.handlePhysics(this, deltaPos, newPosition, newVelocityOut);
+                } else {
+                    newVelocityOut = deltaPos;
```

Comment:
Shouldn't you set it to `getVelocity()` instead?

Reply:
Look at the `hasPhysics` branch: this variable is being passed to `handlePhysics` to be populated there.

Reply:
`hasPhysics` branch? Where is it?
I meant that `newVelocityOut` should probably be set to `getVelocity()` because the vector is copied later

Reply:
I was talking about if branch 2 lines up :)

Reply:
Oh right. I am saying this because `deltaPos` represents the movement delta and not the velocity out

Reply:
Yes, but later we're setting `velocity` to `deltaPos`, multiplied by `tps`, so it seems absolutely correct.

Reply:
My bad, forgot about it